src/status_im/contexts/chat/messenger/photo_selector/effects.cljs

```diff
@@ -80,9 +81,47 @@
                        (callback @albums))))))
               (callback @albums)))))))))
 
+(defn- get-album-cover-uri
+  [photos]
+  (get-in (first photos) [:node :image :uri]))
+
+(defn- get-album-info
+  [album]
+  ;; Easy promise chains without (.then #(...))
+  (p/->> (cameraroll/promise-get-photos
+          {:first      1
+           :groupTypes "albums"
+           :groupName  (:title album)
+           :assetType  "photos"})
+         :edges
+         (get-album-cover-uri)
+         (assoc album :uri)))
+
+(defn- p-get-albums
+  []
+  ;; Using an "async/await-like" pattern with promesa's let
+  (p/let [smart-album-cover-uri (p/-> (cameraroll/promise-get-photos
+                                       {:first      1
+                                        :groupTypes "All"
+                                        :assetType  "Photos"})
+                                      :edges
+                                      get-album-cover-uri)
+          albums                (cameraroll/promise-get-albums {:assetType "Photos"})
+          album-previews        (p/->> (p/all (map get-album-info albums))
+                                       (sort-by :title))]
```

Comment:
"synchronous-looking" async, similar to javascript's `async/await` with `p/let`. It can contain both promises and sync values and is basically syntactic sugar on top of a chain of `(.then ...).